### PR TITLE
fix: incorrect usages of error types

### PR DESCRIPTION
**Pull request by saikumarrs**
## Description of the change

Fixed all the instances of incorrect usage of error types.

Also, removed all the usages of `TRANSFORMER_METRIC` as it is already removed.

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

N/A

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by mihir-4116 — APPROVED** [review]

File: src/cdk/v2/bindings/default.js [hunk 1]
```diff
@@ -26,11 +29,18 @@ function isValidEventType(event) {
 
 function assert(val, message, status, statTags, destination) {
```
Comment (koladilip) [inline src/cdk/v2/bindings/default.js:30]: We should remove from definition too right?

File: src/cdk/v2/bindings/default.js [hunk 2]
```diff
@@ -26,11 +29,18 @@ function isValidEventType(event) {
...
   if (!val) {
-    throw new InstrumentationError(message, status, statTags, destination);
+    throw new InstrumentationError(message);
+  }
+}
+
+function assertConfig(val, message, status, statTags, destination) {
```
Comment (koladilip) [inline src/cdk/v2/bindings/default.js:36]: We should remove from definition too right?

File: src/v0/destinations/canny/util.js [hunk 3]
```diff
@@ -16,55 +16,48 @@ const tags = require("../../util/tags");
...
+      {
+        [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(response.data?.status)
+      },
+      response.data?.error
     );
-  } catch (error) {
-    throw new TransformationError("Unable to retrieve userid from Canny");
   }
+  return (
+    response?.response?.data?.data?.id || response?.response?.data?.id || null
```
Comment (koladilip) [inline src/v0/destinations/canny/util.js:59]: || null required?

File: src/v0/destinations/eventbridge/transform.js [hunk 4]
```diff
@@ -26,30 +26,22 @@ function getResouceList(config) {
...
-      throw new ConfigurationError(
-        "EventBridge: received empty config, dropping event",
-        400
-      );
-    }
-  } catch (error) {
-    throw new TransformationError(
-      error.message || "EventBridge: Unknown error"
-    );
+  if (event.destination && event.destination.Config) {
```
Comment (koladilip) [inline src/v0/destinations/eventbridge/transform.js:29]: if (event.destination?.Config) {

File: src/v0/destinations/gainsight_px/util.js [hunk 5]
```diff
@@ -83,7 +91,16 @@ const createAccount = async (payload, Config) => {
     if (response && response.status === 201) {
       return { success: true, err: null };
     }
-    throw new TransformationError("invalid response while creating account");
+
+    const status = response ? response.status : undefined;
```
Comment (koladilip) [inline src/v0/destinations/gainsight_px/util.js:95]: const status = response?.status || some default error status

File: src/v0/destinations/google_adwords_enhanced_conversions/transform.js [hunk 6]
```diff
@@ -104,12 +104,7 @@ const processTrackEvent = async (metadata, message, destination) => {
...
-  let payload;
-  try {
-    payload = constructPayload(message, updatedMapping);
-  } catch (e) {
-    throw new TransformationError(`${e.message} for ${event} event.`);
-  }
+  const payload = constructPayload(message, updatedMapping);
 
   payload.partialFailure = true;
   if (!payload.conversionAdjustments[0].userIdentifiers) {
```
Comment (koladilip) [inline src/v0/destinations/google_adwords_enhanced_conversions/transform.js:110]: payload.conversionAdjustments[0]?.userIdentifiers

File: src/v0/destinations/trengo/transform.js [hunk 7]
```diff
@@ -233,7 +233,7 @@ const ticketBuilderTrengo = async (message, destination, identifer, extIds) => {
         };
         subjectLine = hTemplate(templateInput).trim();
       } catch (err) {
-        throw new TransformationError(
+        throw new InstrumentationError(
           `Error occurred in parsing event template for ${message.event}`,
           400
```
Comment (koladilip) [inline src/v0/destinations/trengo/transform.js:238]: status required?

File: src/cdk/v2/bindings/default.js (hunk 1 above)
Comment (saikumarrs) [reply src/cdk/v2/bindings/default.js]: Fixed.

File: src/cdk/v2/bindings/default.js (hunk 2 above)
Comment (saikumarrs) [reply src/cdk/v2/bindings/default.js]: Fixed.

File: src/v0/destinations/eventbridge/transform.js (hunk 4 above)
Comment (saikumarrs) [reply src/v0/destinations/eventbridge/transform.js]: Fixed.

File: src/v0/destinations/gainsight_px/util.js (hunk 5 above)
Comment (saikumarrs) [reply src/v0/destinations/gainsight_px/util.js]: Fixed.

File: src/v0/destinations/google_adwords_enhanced_conversions/transform.js (hunk 6 above)
Comment (saikumarrs) [reply src/v0/destinations/google_adwords_enhanced_conversions/transform.js]: Fixed.

File: src/v0/destinations/trengo/transform.js (hunk 7 above)
Comment (saikumarrs) [reply src/v0/destinations/trengo/transform.js:238]: Fixed.

**Review by ujjwal-ab — APPROVED** [review]

**Review by koladilip — APPROVED** [review]

**Pull request merged**